### PR TITLE
Obsolete ReferenceWrapper.Name in favor of FriendlyName

### PR DESCRIPTION
**Pull request by nulltoken**
@jamill rightfully pointed at the fact that `Name` was maybe not explicit enough for a `Branch` or a `Tag`.

This PR is proposal to work around this by explicitly stating that this is a shortened version of the complete canonical name.

Thoughts?

### CODE REVIEW

**Comment by jamill** [comment]
To throw out another option, what about naming property `FriendlyName`? Otherwise, I think `ShortName` also captures what the property does.

**Comment by carlosmn** [comment]
IIRC git also calls it the "human-friendly name". In libgit2 itself we call it "shorthand" which to me conveys more of the semantic meaning.

**Comment by nulltoken** [comment]
Let's go with `FriendlyName`, then.

**Comment by nulltoken** [comment]
Friendlier?

File: LibGit2Sharp/ReferenceWrapper.cs [hunk 1]
```diff
@@ -52,11 +52,20 @@ public virtual string CanonicalName
...
+        /// </summary>
+        public virtual string FriendlyName
+        {
+            get { return Shorten(); }
+        }
+
+        /// <summary>
         /// Gets the name of this reference.
         /// </summary>
+        [Obsolete("This property will be removed in the next release. Please use ShortName instead.")]
```
Comment (jamill) [inline LibGit2Sharp/ReferenceWrapper.cs]: ShortName -> FriendlyName

Comment (nulltoken) [inline LibGit2Sharp/ReferenceWrapper.cs]: Thanks! Fixed.

**Comment by nulltoken** [comment]
Published as NuGet pre-release package `LibGit2Sharp.0.22.0-pre20150415174523`

**Pull request merged**